app/code/Magento/Catalog/Model/Product/Price/BasePriceStorage.php

```diff
@@ -146,6 +177,16 @@ public function update(array $prices)
             }
         }
 
+        try {
```

Comment:
I would not add here try / catch. Cause if such attribute do not exists, exception will be much earlier, already on $this->getPricePersistence() method